pod/perlre.pod

```diff
@@ -1193,54 +1257,68 @@ would match the same as C</(Y) ( (X) \g3 \g1 )/x>.  This allows you to
-to that name assumes the leftmost defined group.  Named groups count in
-absolute and relative numbering, and so can also be referred to by those
-numbers.
-(It's possible to do things with named capture groups that would otherwise
-require C<(??{})>.)
-
-Capture group contents are dynamically scoped and available to you outside the
-pattern until the end of the enclosing block or until the next successful
-match in the same scope, whichever comes first.
-See L<perlsyn/"Compound Statements"> and
+When different groups within the same pattern have the same name, any
+reference to that name assumes the leftmost defined group. Named groups
+count in absolute and relative numbering, and so can also be referred to
+by those numbers. (It's possible to do things with named capture groups
+that would otherwise require C<(??{})>.)
+
+Inside of a I<new> match you can access the contents of a capture group
+from the most recent previous successful match by using the same
+variables as you would outside of a pattern. Thus in the second match in
+the following example
+
+    "foo" =~ /(foo)/;
+    "barfoobar" =~ /(bar)$1\1/;
+
+the C<$1> refers to content of the first capture group from the
+B<previous> match, and the C<\1> refers to the contents of the first
+capture group of the B<current> match.
+
+Capture group contents are dynamically scoped and available to you
+outside the pattern until the end of the enclosing block or until the
+next successful match in the same scope, whichever comes first. See
+L<perlsyn/"Compound Statements"> and
 L<perlvar/"Scoping Rules of Regex Variables"> for more details.
 
-You can access the contents of a capture group by absolute number (using
-C<"$1"> instead of C<"\g1">, I<etc>); or by name via the C<%+> hash,
-using C<"$+{I<name>}">.
+=head4 Use of curly braces in backreference notation
 
-Braces are required in referring to named capture groups, but are optional for
-absolute or relative numbered ones.  Braces are safer when creating a regex by
-concatenating smaller strings.  For example if you have C<qr/$a$b/>, and C<$a>
-contained C<"\g1">, and C<$b> contained C<"37">, you would get C</\g137/> which
-is probably not what you intended.
+Curly braces are I<required> in referring to named capture groups, but
+are optional for absolute or relative numbered ones. Braces are safer
+when creating a regex by concatenating smaller strings. For example if
+you have C<qr/$x$y/>, and C<$x> contained C<"\g1">, and C<$y>
+contained C<"37">, you would get C</\g137/> which is probably not what
+you intended and will likely throw an exception during compilation (unless
+the pattern actually contained 137 capture buffers).
 
 If you use braces, you may also optionally add any number of blank
 (space or tab) characters within but adjacent to the braces, like
-S<C<\g{ -1 }>>, or S<C<\k{ I<name> }>>.
+S<C<\g{ -1 }>>, or S<C<\k{ I<name> }>>. Notice that C<\g{ name }> and
```

Comment:
I think it would be better to just stick to `\g` in examples after explaining what `\k` is once, since it's rarely used.